index.js

```diff
@@ -19,215 +19,49 @@
-  {location: [37.789251178427776, -122.4013303460217], id: 2},
-  {location: [37.786862920252986, -122.40475531334141], id: 3},
-  {location: [37.78861431712821, -122.40505751634022], id: 4},
-  {location: [37.79060449046487, -122.40556118800487], id: 5},
-  {location: [37.790047247333675, -122.4088854209916], id: 6},
-  {location: [37.79275381746233, -122.4091876239904], id: 7},
-  {location: [37.795619489534374, -122.40989276432093], id: 8},
-  {location: [37.79792786675678, -122.41049717031848], id: 9},
-  {location: [37.80031576728801, -122.4109001076502], id: 10},
-  {location: [37.79920142331301, -122.41916032295062], id: 11}
-];
+var contents = {
+  'Getting Started': require('./content/getting-started/getting-started.js'),
+  'Scatterplot': require('./content/scatterplot/scatterplot.js'),
+  'Draggable Points': require('./content/draggable-points/draggable-points'),
+  'Third Party Overlays': require('./content/third-party-overlay/third-party-overlay'),
+}
 
-var pointIds = initialPoints[initialPoints.length - 1].id;
+var titles = Object.keys(contents);
 
-var Docs = React.createClass({
+var sidebar = document.getElementById('sidebar');
 
-  getInitialState: function getInitialState() {
-    var normal = d3.random.normal();
-    function wiggle(scale) {
-      return normal() * scale;
-    }
-    return {
-      map: {
-        latitude: 37.78,
-        longitude: -122.45,
-        zoom: 11,
-        mapStyle: stamenMapStyle,
-        width: 450,
-        height: 450,
-        startDragLatLng: null
-      },
-      draggablePoints: Immutable.fromJS(initialPoints),
-      locations: Immutable.fromJS(d3.range(2000).map(function _map() {
-        return [37.788 + wiggle(0.01), -122.408 + wiggle(0.01)];
-      }))
-    };
-  },
 
-  _onAddPoint: function _onAddPoint(_location) {
-    var points = this.state.draggablePoints.push(new Immutable.Map({
-      location: new Immutable.List(_location),
-      id: ++pointIds
-    }));
-    this.setState({draggablePoints: points});
-  },
+// Fill out the sidebar menu items from the contents list.
+for (var i = 0; i < titles.length; i++) {
+
+  var title = titles[i];
+  var component = contents[title];
 
-  _onUpdatePoint: function _onUpdatePoint(opt) {
-    var index = this.state.draggablePoints.findIndex(function filter(p) {
-      return p.get('id') === opt.key;
-    });
-    var point = this.state.draggablePoints.get(index);
-    point = point.set('location', new Immutable.List(opt.location));
-    var points = this.state.draggablePoints.set(index, point);
-    this.setState({draggablePoints: points});
-  },
+  var menuitem = document.createElement('p');
+  menuitem.className = 'menuitem';
+  menuitem.innerText = title;
 
-  _onChangeViewport: function _onChangeViewport(opt) {
-    this.setState({map: assign({}, this.state.map, opt)});
-  },
+  menuitem.addEventListener('click', (function(t, c) {
+    return function() {
+      setLocationHash(t, {replace: true});
```

Comment:
I haven't used it myself but I've heard [react-router](https://github.com/rackt/react-router) is good for this kind of thing